benchmarks/battleship/benchmark/src/main/java/com/android/designcompose/benchmark/battleship/benchmark/ClickPlayfield.kt

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.android.designcompose.benchmark.battleship.benchmark
+
+import androidx.benchmark.macro.ExperimentalMetricApi
+import androidx.benchmark.macro.TraceSectionMetric
+import androidx.benchmark.macro.TraceSectionMetric.Mode
+import androidx.benchmark.macro.junit4.MacrobenchmarkRule
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.uiautomator.By
+import androidx.test.uiautomator.Until
+import com.android.designcompose.DCTraces
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+/**
+ * This is a rough example of a benchmark, though one that doesn't focus on any behavior well.
+ * Essentially it benchmarks a large recomposition, by clicking the Opponent's Playfield button,
+ * which causes the full screen to re-draw.
+ */
+@OptIn(ExperimentalMetricApi::class)
+@RunWith(AndroidJUnit4::class)
+class ClickPlayfield {
+
+    @get:Rule val benchmarkRule = MacrobenchmarkRule()
+
+    @Test
+    fun changePlayers() =
+        benchmarkRule.measureRepeated(
+            packageName = PACKAGE_NAME,
+            metrics =
+                listOf(
+                    TraceSectionMetric(DCTraces.DESIGNDOCINTERNAL),
+                    TraceSectionMetric(DCTraces.DESIGNVIEW),
```

Comment:
Is there no `Mode.Sum` for these two trace types because you're assuming that the first call to them is the root node, so we can ignore all the children? Also, I think I asked this already but you can tell me again how it works when trace calls are nested? `DesignDocInternal` calls `DesignView`, which calls `DesignFrame` which calls `subscribe`, etc.

Reply:
Yeah, I was thinking that they should only be called once, but I dunno. It'd be best to experiment to see what works and what doesn't.
Nested trace calls don't influence each other, they record and report individually.